src/state_machine.rs

```diff
@@ -346,6 +359,83 @@ impl StateMachine {
         }
     }
 
+    /// Query for a result, or yield: Err(NothingAvailable), Err(Disconnected).
+    #[cfg(feature = "use-mock-crust")]
+    pub fn try_step(&mut self, outbox: &mut EventBox) -> Result<(), TryRecvError> {
+        use itertools::Itertools;
+        use maidsafe_utilities::SeededRng;
+        use rand::Rng;
+        use std::borrow::Borrow;
+        use std::iter::{self, Iterator};
+
+        if self.is_running {
+            let mut events = Vec::new();
+            while let Ok(category) = self.category_rx.try_recv() {
+                match category {
+                    MaidSafeEventCategory::Routing => {
+                        if let Ok(action) = self.action_rx.try_recv() {
+                            events.push(EventType::Action(Box::new(action)));
+                        } else {
+                            return Ok(self.apply_transition(Transition::Terminate, outbox));
+                        }
+                    }
+                    MaidSafeEventCategory::Crust => {
+                        if let Ok(crust_event) = self.crust_rx.try_recv() {
+                            events.push(EventType::CrustEvent(crust_event));
+                        } else {
```

Comment:
yes, I agree.
the current mock_crust does need to be `work asynchronously` to fully simulate the behaviour of production code.